src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -246,17 +246,22 @@ KernelsData DetectionOutputKernelRef::GetKernelsData(const Params& params, const
                                        MakeJitConstant("NUM_PRIOR_BLOCKS", num_score_block)});
             }
         } else if (i == 1) {
-             if (detectOutParams.detectOutParams.decrease_label_id) {
+            if (detectOutParams.detectOutParams.decrease_label_id) {
+                // Always use local memory since LWS size is 1x1x16 (16 WI * 100 (stack size) * 4 (int size) = 6.25 KB of SLM memory)
+                cldnnJit.AddConstant(MakeJitConstant("USE_LOCAL_MEMORY_FOR_STACK", "true"));
                 cldnnJit.AddConstants({MakeJitConstant("DO_STAGE_" + std::to_string(i) + "_MXNET", "true"),
                                        MakeJitConstant("LOCAL_WORK_NUM", dispatchData.lws[2]),
                                        MakeJitConstant("PARTITION_STEP", GetPartitionStep(dispatchData.lws[2]))});
-             } else {
+            } else {
+                // Limit local memory usage by 8*16 LWS (128 WI * 100 (stack size) * 4 (int size) = 50 KB of SLM memory)
+                if (dispatchData.lws[0] * dispatchData.lws[1] <= 8 * 16)
```

Comment:
Don't we need to have a comparison with available local mem size somewhere here to handle different SKUs?

Reply:
Thanks, fixed